public/components/common/seach/queryBar.tsx

```diff
@@ -9,28 +9,83 @@
  * GitHub history for details.
  */
 
-import React from 'react';
+import React, {Fragment, useEffect, useState, useCallback} from 'react';
 import {
+  EuiButtonEmpty,
   EuiFieldSearch,
   EuiFlexItem,
+  EuiSuggest,
+  EuiFlexGroup,
+  EuiSelectableTemplateSitewide,
+  EuiSelectableTemplateSitewideOption,
+  EuiSuggestItem,
+  EuiSuggestItemProps
 } from '@elastic/eui';
+import useDebounce from 'react-use/lib/useDebounce'
 import {
   IQueryBarProps
 } from './search';
 import {
   RAW_QUERY
 } from '../../../common/constants/explorer';
+import { _termValuesToQuery } from '@elastic/eui/src/components/search_bar/query/ast_to_es_query_dsl';
+//import {getQuerySuggestions} from './autocomplete_service'
 
-export function QueryBar(props: IQueryBarProps) {
 
+const commands: EuiSuggestItemProps[] =[
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'dedup'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'eval'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'fields'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'rename'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'search'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'sort'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'stats'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'where'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'head'},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'top'},
+  {type: {iconType: 'search', color: 'tint1'}, label: 'source'}
+]
+
+export function QueryBar(props: IQueryBarProps) {
+  
   const {
     query,
     handleQueryChange,
     handleQuerySearch
   } = props;
 
+  const [value, setValue] = useState('');
+  const [inputValue, setInputValue] = useState(value);
+
+  const splittedModel = value.split(',');
+  const prefix = splittedModel[Math.max(0, splittedModel.length - 1)]
+  const suggestions = commands.filter(
+    ({ label }) => label.startsWith(inputValue) && prefix !== label
+  )
+
+  const onItemClick = useCallback(
+    ({ label }) => {
+      setInputValue(value.substring(0, value.lastIndexOf(prefix)) + label);
+    },
+    [value, prefix]
+  );
+
+  const onInputChange = useCallback((target) => {
+    setInputValue(target.value);
+    handleQueryChange(target.value);
+  }, [])
+
+  const append = <EuiButtonEmpty>PPL</EuiButtonEmpty>
```

Comment:
Do we want this to be clickable? also you might want to adjust the size of `PPL`, default size might be larger than other components

Reply:
nah, i was just following the [EuiSuggest documentation](https://elastic.github.io/eui/#/forms/suggest). it's unnecessary since the placeholder indicates that it should be PPL, just thought it would look clearer at the time. will be taking it out